src/pages/HomePage/pages/MarketPlace/components/forms/ProductForm.tsx

```diff
@@ -131,6 +126,93 @@ export function ProductForm({
                                   {errors.gallery[index]?.image}
                                 </p>
                               )}
+                            {/* size and stocks */}
+                            {values.manage_stock === "yes" && (
+                              <div>
+                                <FieldArray
+                                  name={`gallery[${index}].stock_management`}
```

Comment:
also the FieldArray is really hard to follow please create a new component below this form with the fieldArray and call it inside the main form component